File: components/MongoMigration/version/1.1.1/tag.js
const debug = require('debug')(`linto:components:MongoMigration:controllers:version:1.1.1:tags`)

const collections_name = 'tags'

const tag = {
  name: 'tag', // podcast
  color: 'color',
  organizationId: 'organizationId',
  categoryId: 'categoryId' //vocal
}

const initDb = require(`${process.cwd()}/components/MongoMigration/controllers/migration/init`)
const dropDb = require(`${process.cwd()}/components/MongoMigration/controllers/migration/drop`)

module.exports = {
  async up(db) {
    await initDb(db, collections_name)
  },

  async down(db) {
    await dropDb(db, collections_name)
  }
}

File: studio-app/lib/data/tag.dart
import 'package:flutter/material.dart';

class TagCategory {
  final String _id;
  String get id => _id;
  final String _name;
  String get name => _name;
  final List<Tag> _tags;
  List<Tag> get tags => _tags;
  final Color _color;
  Color get color => _color;

  TagCategory(String id, String name, List<Tag> tags, Color color)
      : _id = id, _name = name, _tags = tags, _color = color;
}

class Tag {
  final String _id;
  String get id => _id;
  final String _name;
  String get name => _name;

  Tag(String id, String name) : _id = id, _name = name;
}
